### PR TITLE
Complete overhaul of the library

### PR DESCRIPTION
**Pull request by 9b**
- Added proper setup files and documentation
- Package is PIP-friendly and can be submitted via Twine
- Added structure to the install package to ease future updates
- Included helper functions to page and recurse through results, and provide more content
- Abstracted API endpoints to friendly names
- Added a logging interface
- Centralized requesting logic to be more extensible

### CODE REVIEW

**Comment by andrew-morris** [comment]
Can commit to having this reviewed this week.

**Comment by andrew-morris** [comment]
@9b can you remove boilerplate references to google-alerts plz

File: greynoise/__init__.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+def valid_date(date):
+    """Check the input date and ensure it matches the format."""
+    import datetime
+    try:
+        datetime.datetime.strptime(date, '%Y-%m-%d')
+    except ValueError:
+        raise ValueError("Incorrect data format, should be YYYY-MM-DD")
+
+
+def valid_ip(ip_address, strict=True):
```
Comment (andrew-morris) [inline greynoise/__init__.py]: should we return false if RFC 1918?

Comment (9b) [reply greynoise/__init__.py]: Technically, it's an IP.

**Pull request merged**